src/transform/FooterReadMore.js

```diff
@@ -0,0 +1,349 @@
+ * @param {!Document} document
+ * @return {void}
+ */
+const fetchReadMore = (baseURL, title, showReadMoreHandler, containerID, saveButtonClickHandler,
+  titlesShownHandler, document) => {
+  const xhr = new XMLHttpRequest() // eslint-disable-line no-undef
+  xhr.open('GET', readMoreQueryURL(title, baseURL), true)
+  xhr.onload = function() {
+    if (xhr.readyState === 4) {
+      if (xhr.status === 200) {
+        showReadMoreHandler(JSON.parse(xhr.responseText).query.pages, document, containerID,
+          saveButtonClickHandler, titlesShownHandler)
+      } else {
+        fetchErrorHandler(xhr.statusText)
+      }
+    }
+  }
+  xhr.onerror = function(e) {
+    fetchErrorHandler(xhr.statusText)
+  }
+  xhr.send(null)
+}
+
+/**
+ * Updates save button text for saved state.
+ * @param {!HTMLDivElement} button
+ * @param {!string} title
+ * @param {!Boolean} isSaved
+ * @return {void}
+ */
+const updateSaveButtonText = (button, title, isSaved) => {
+  const text = isSaved ? _savedForLaterString : _saveForLaterString
+  button.innerText = text
+  button.title = text
+}
+
+/**
+ * Updates save button bookmark icon for saved state.
+ * @param {!HTMLDivElement} button
+ * @param {!string} title
+ * @param {!Boolean} isSaved
+ * @return {void}
+ */
+const updateSaveButtonBookmarkIcon = (button, title, isSaved) => {
+  const unfilledClass = 'pagelib_footer_readmore_bookmark_unfilled'
+  const filledClass = 'pagelib_footer_readmore_bookmark_filled'
+  button.classList.remove(unfilledClass)
+  button.classList.remove(filledClass)
+  button.classList.add(isSaved ? filledClass : unfilledClass)
+}
+
+/**
+ * Updates save button text and bookmark icon for saved state.
+ * @param {!string} title
+ * @param {!Boolean} isSaved
+ * @param {!Document} document
+ * @return {void}
+*/
+const setTitleIsSaved = (title, isSaved, document) => {
+  const saveButton = document.getElementById(`${_saveButtonIDPrefix}${title}`)
+  updateSaveButtonText(saveButton, title, isSaved)
+  updateSaveButtonBookmarkIcon(saveButton, title, isSaved)
+}
+
+/**
+ * Adds 'Read more' to 'containerID' element.
+ * @param {!Document} document
+ * @param {?string} baseURL
+ * @param {!string} title
+ * @param {!string} saveForLaterString
+ * @param {!string} savedForLaterString
+ * @param {!string} containerID
+ * @param {SaveButtonClickHandler} saveButtonClickHandler
+ * @param {TitlesShownHandler} titlesShownHandler
+ */
+const add = (document, baseURL, title, saveForLaterString, savedForLaterString, containerID,
+  saveButtonClickHandler, titlesShownHandler) => {
+  _saveForLaterString = saveForLaterString
+  _savedForLaterString = savedForLaterString
+  fetchReadMore(
+    baseURL, title, showReadMore, containerID, saveButtonClickHandler, titlesShownHandler, document
+  )
+}
+
+/**
+ * Sets heading element string.
+ * @param {!string} headingString
+ * @param {!string} headingID
+ * @param {!Document} document
+ */
+const setHeading = (headingString, headingID, document) => {
```

Comment:
Consider consistently placing the document / window parameter in the same position in each method signature unless there's a good reason to deviate.